tests/wpt/grouping_formatter.py

```diff
@@ -17,16 +17,22 @@ def __init__(self):
         self.need_to_erase_last_line = False
         self.current_display = ""
         self.running_tests = {}
-        self.last_test_finished = "Running tests..."
         self.test_output = collections.defaultdict(str)
         self.subtest_failures = collections.defaultdict(list)
         self.test_failure_text = ""
         self.tests_with_failing_subtests = []
         self.interactive = os.isatty(sys.stdout.fileno())
 
-        # iTerm2 doesn't support the terminal codes used to erase previous lines,
-        # so only print one line and rely only on backspace characters.
-        self.one_line = os.environ.get("TERM_PROGRAM", "") == "iTerm.app"
+        # TODO(mrobinson, 8313): We need to add support for Windows terminals here.
+        if self.interactive:
+            try:
+                import blessings
+                self.terminal = blessings.Terminal()
+                self.clear_eol = self.terminal.clear_eol
+                self.move_up = self.terminal.move_up
+            except:
```

Comment:
try and except with no condition is never a good idea; even `except Exception:` is better because it won't catch `KeyboardInterrupt`. In this case I agree that splitting into two blocks is a better idea.